--- submodule/acme_copy/acme/dnsplugin/plugin.go	2024-05-13 12:52:52.500124415 +0800
+++ submodule/acme/acme/dnsplugin/plugin.go	2024-05-13 12:53:04.844254546 +0800
@@ -11,7 +11,7 @@
 	"github.com/go-acme/lego/v4/challenge"
 	"github.com/go-acme/lego/v4/challenge/dns01"
 	"github.com/hashicorp/go-plugin"
-	dnspluginproto "github.com/vancluever/terraform-provider-acme/v2/proto/dnsplugin/v1"
+	dnspluginproto "github.com/myklst/terraform-provider-acme/v2/proto/dnsplugin/v1"
 	"google.golang.org/grpc"
 	"google.golang.org/protobuf/types/known/durationpb"
 )
